From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 16 Jul 2023 13:01:20 +0200
Subject: [PATCH] Create unit test classes for Block and Item

Creates unit test classes `PerBlockTest` and `PerItemTest` for tests related to respectively the `Block` and `Item` classes.
Does not add any tests: these are added in subsequent patches.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/test/java/org/fiddlemc/fiddle/block/PerBlockTest.java b/src/test/java/org/fiddlemc/fiddle/block/PerBlockTest.java
new file mode 100644
index 0000000000000000000000000000000000000000..d7d99910630a7e8267352b734bc2eee6c91982f6
--- /dev/null
+++ b/src/test/java/org/fiddlemc/fiddle/block/PerBlockTest.java
@@ -0,0 +1,18 @@
+// Fiddle - tests related to net.minecraft.world.level.block.Block
+
+package org.fiddlemc.fiddle.block;
+
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.world.level.block.Block;
+import org.bukkit.support.AbstractTestingBase;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class PerBlockTest extends AbstractTestingBase {
+
+    private Iterable<Block> getBlocks() {
+        return BuiltInRegistries.BLOCK;
+    }
+
+}
diff --git a/src/test/java/org/fiddlemc/fiddle/item/PerItemTest.java b/src/test/java/org/fiddlemc/fiddle/item/PerItemTest.java
new file mode 100644
index 0000000000000000000000000000000000000000..911160beec65ff1d865e068f8ad817d6b533ebda
--- /dev/null
+++ b/src/test/java/org/fiddlemc/fiddle/item/PerItemTest.java
@@ -0,0 +1,18 @@
+// Fiddle - tests related to net.minecraft.world.item.Item
+
+package org.fiddlemc.fiddle.item;
+
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.world.item.Item;
+import org.bukkit.support.AbstractTestingBase;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class PerItemTest extends AbstractTestingBase {
+
+    private Iterable<Item> getItems() {
+        return BuiltInRegistries.ITEM;
+    }
+
+}
